user/2fa/remove.ts

```diff
@@ -0,0 +1,41 @@
+import { gracely } from "gracely"
+import { isoly } from "isoly"
+import { userwidgets } from "@userwidgets/model"
+import { http } from "cloudly-http"
+import { Context } from "../../Context"
+import { router } from "../../router"
+
+export async function remove(request: http.Request, context: Context): Promise<http.Response.Like | any> {
+	let result: userwidgets.User | gracely.Error
+	const credentials = gracely.Error.is(context.authenticator)
+		? context.authenticator
+		: await context.authenticator.authenticate(request, "token")
+	const entityTag = request.header.ifMatch?.at(0)
+	if (!request.parameter.email)
+		result = result = gracely.client.invalidPathArgument(
+			"/user/:email/2fa",
+			"email",
+			"string",
+			"email missing from path."
+		)
+	else if (gracely.Error.is(context.users))
+		result = context.users
+	else if (!entityTag)
+		result = gracely.client.missingHeader("If-Match", "If-Match header must contain an entity tag.")
+	else if (entityTag != "*" && !isoly.DateTime.is(entityTag))
+		result = gracely.client.malformedHeader("ifMatch", "If-Match header must contain a valid entity tag.")
+	else if (gracely.Error.is(credentials))
+		result = credentials
+	else if (!credentials)
+		result = gracely.client.unauthorized()
+	else if (
+		!userwidgets.User.Permissions.organizations(credentials.permissions).every(id =>
+			userwidgets.User.Permissions.check(credentials.permissions, id, "user.admin")
+		)
+	)
+		result = gracely.client.unauthorized()
+	else
+		result = await context.users.remove2fa(request.parameter.email)
```

Comment:
This function should require the entity tag check in the else if chain. without the entity tag its impossible to determine mid air collations. Due to the nature of what this endpoint is supposed to do maybe it inst necessary. either remove the checks for the entity tag or make sure you use the tag all the way down into the DO.

Reply:
It doesn't seem to be necessary for a delete endpoint, so I'll remove the checks.